site/content/docs/reference.md

```diff
@@ -408,3 +411,140 @@ spec:
 ```
 
 _ref: [pkg/apis/v1alpha1/cluster_template.go](../../../pkg/apis/v1alpha1/cluster_template.go)_
+
+
+### Runnable
+
+A `Runnable` object declares the intention of having immutable objects
+submitted to Kubernetes according to a template (via ClusterRunTemplate)
+whenever any of the inputs passed to it changes. i.e., it allows us to provide
+a mutable spec that drives the creation of immutable objects whenever that spec
+changes.
+
+
+```yaml
+apiVersion: carto.run/v1alpha1
+kind: Runnable
+metadata:
+  name: test-runner
+spec:
+  # data to be made available to the template of ClusterRunTemplate
+  # that we point at.
+  #
+  # this field takes as value an object that maps strings to values of any
+  # kind, which can then be reference in a template using jsonpath such as
+  # `$(runnable.spec.inputs.<key...>)$`.
+  #
+  # (required)
+  #
+  inputs:
+    git-url: https://github.com/vmware-tanzu/cartographer
+    git-revision: f397a1f349146442c46051099d7e32ebaec79623
+
+    foo: [{bar: 123}]
```

Comment:
I'm not super sure about this one - I wanted to showcase that we take for inputs any json type

https://github.com/vmware-tanzu/cartographer/blob/1afa331490ebd0d2bb258f074aee480fb8477efe/pkg/apis/v1alpha1/runnable.go#L56-L61

but ... the example looks a bit strange imo

Reply:
maybe we could make it instead:

```
spec:
  inputs:
    serviceAccount: bla
    params:
       - name: foo
          value: bar
```